pkg/queue/cluster_queue_test.go

```diff
@@ -86,6 +90,36 @@ func Test_PushOrUpdate(t *testing.T) {
 						Reason: kueue.WorkloadEvictedByPodsReadyTimeout,
 						Status: metav1.ConditionTrue,
 					}).
+					Condition(metav1.Condition{
+						Type:   kueue.WorkloadRequeued,
+						Status: metav1.ConditionFalse,
+					}).
+					Obj()),
+			},
+		},
+		"should wait for Requeued=true after backoff waiting time before push to heap": {
+			workload: wlBase.Clone().
+				Condition(metav1.Condition{
+					Type:   kueue.WorkloadEvicted,
+					Reason: kueue.WorkloadEvictedByPodsReadyTimeout,
+					Status: metav1.ConditionTrue,
+				}).
+				Condition(metav1.Condition{
+					Type:   kueue.WorkloadRequeued,
+					Status: metav1.ConditionFalse,
+				}),
+			wantInAdmissibleWorkloads: map[string]*workload.Info{
+				"default/workload-1": workload.NewInfo(wlBase.Clone().
+					ResourceVersion("1").
+					Condition(metav1.Condition{
+						Type:   kueue.WorkloadEvicted,
+						Reason: kueue.WorkloadEvictedByPodsReadyTimeout,
+						Status: metav1.ConditionTrue,
+					}).
+					Condition(metav1.Condition{
+						Type:   kueue.WorkloadRequeued,
+						Status: metav1.ConditionFalse,
+					}).
 					Obj()),
 			},
 		},
```

Comment:
Could you add a case to verify the situation of changed Requeued Condition?

Reply:
Done